client.go

```diff
@@ -235,10 +238,6 @@ type Client struct {
 // single goroutine) or hub methods (for concurrent programs, for example web
 // servers).
 func NewClient(options ClientOptions) (*Client, error) {
-	if options.TracesSampleRate != 0.0 && options.TracesSampler != nil {
```

Comment:
I removed this error, as `TracesSampler` only takes priority over `TracesSampleRate`